src/SecurityAdvisoriesCheck.php

```diff
@@ -15,21 +16,63 @@ class SecurityAdvisoriesCheck extends Check
     /** @var array<string> */
     protected array $ignoredPackages = [];
 
+    protected int $retryTimes = 5;
+
+    public PackagistClient $packagistClient;
+
+    public function __construct(?PackagistClient $packagistClient = null)
+    {
+        parent::__construct();
+
+        $this->packagistClient = $packagistClient ?? new PackagistClient(new Client(), new PackagistUrlGenerator());
+    }
+
+    public function retryTimes(int $times): self
+    {
+        $this->retryTimes = $times;
+
+        return $this;
+    }
+
     public function run(): Result
     {
         $packages = $this->getInstalledPackages();
 
-        $advisories = retry(5, function () use ($packages) {
-            return $this->getAdvisories($packages);
-        }, sleepMilliseconds: 2 * 1000);
+        $packagistGatewayErrorCount = 0;
+        $lastNonGatewayPackagistError = null;
+
+        try {
```

Comment:
Is there a way to make this more readable? Maybe by refactoring the closures to named function of their own?

Reply:
I've refactored closures and conditions to separate methods